### PR TITLE
AP_UAVCAN: Broadcast ArmingStatus regularly

### PR DESCRIPTION
**Pull request by WickedShell**
This allows us to regularly broadcast the `ArmingStatus` message for UAVCAN nodes to use to make decisions. I'm looking to use this on AP_Periph nodes for example to dictate what they should be doing.

Validated on a CubeOrange using SLCAN.

### CODE REVIEW

File: libraries/AP_UAVCAN/AP_UAVCAN.cpp [hunk 1]
```diff
@@ -702,25 +709,35 @@ void AP_UAVCAN::rtcm_stream_send()
...
+        switch (hal.util->safety_switch_state()) {
+        case AP_HAL::Util::SAFETY_ARMED:
+            safety_msg.status = ardupilot::indication::SafetyState::STATUS_SAFETY_OFF;
+            break;
+        case AP_HAL::Util::SAFETY_DISARMED:
+            safety_msg.status = ardupilot::indication::SafetyState::STATUS_SAFETY_ON;
+            break;
+        default:
+            // nothing to send
+            return;
```
Comment (tridge) [inline libraries/AP_UAVCAN/AP_UAVCAN.cpp:730]: needs to be a break?

**Review by tridge — CHANGES_REQUESTED** [review]
indentation looks odd?

File: libraries/AP_UAVCAN/AP_UAVCAN.cpp [hunk 2]
```diff
@@ -702,25 +709,35 @@ void AP_UAVCAN::rtcm_stream_send()
 // SafetyState send
 void AP_UAVCAN::safety_state_send()
 {
-    ardupilot::indication::SafetyState msg;
+    ardupilot::indication::SafetyState safety_msg;
```
Comment (tridge) [inline libraries/AP_UAVCAN/AP_UAVCAN.cpp:712]: could be inside the {}

File: libraries/AP_UAVCAN/AP_UAVCAN.cpp (hunk 1 above)
Comment (WickedShell) [reply libraries/AP_UAVCAN/AP_UAVCAN.cpp]: Fixed.

File: libraries/AP_UAVCAN/AP_UAVCAN.cpp (hunk 2 above)
Comment (WickedShell) [reply libraries/AP_UAVCAN/AP_UAVCAN.cpp]: Fixed.

**Comment by WickedShell** [comment]
I've added the `AP_Periph` hook to set the soft armed state. This allows the AP_Periph build to do things like unmount the GPS's SD card if `GPS_RAW_DATA` is set to 2. We still need to do somethings to improve the remounting, but this is much better for the general single flight mapping case.

**Review by tridge — APPROVED** [review]

**Pull request merged**